src/backend/metal/src/command.rs

```diff
@@ -1933,6 +1931,9 @@ impl com::RawCommandBuffer<Backend> for CommandBuffer {
         }
 
         for clear in clears {
+            let pso; // has to live at least as long as all the commands
```

Comment:
What's the problem with leaving the `pso` declaration at the previous location?

Reply:
lifetimes... not really as straightforward as I'd hoped. Basically, anything used in a command has to be declared prior to *all other* commands submitted in the same "batch".